### ISSUE TITLE
rss output is not xml format

### ISSUE THREAD

**Issue by gembes** [backdrop/backdrop-issues#2020]
rss.xml output not in pure xml format, there is html format included..

![error - opera_001](https://cloud.githubusercontent.com/assets/20469101/16862268/7a2866fc-4a72-11e6-84ca-72713d264dc9.png)

**Comment by gembes** [comment]
just add backdrop_exit() and everything goes fine..
file : node.module
line : 2039 
`print $output;`
`backdrop_exit();`


please fix this annoying problem in core.. 
thank

**Comment by serundeputy** [comment]
Thanks @gembes another detailed report.

I've implemented the suggested fix and added a test to check that `rss.xml` is not outputting html.

**Comment by serundeputy** [comment]
this is breaking `NodeFeedTestCase` though.

**Comment by quicksketch** [comment]
Yeah strange, fatal error thrown now. I'll move this to the next release. Seems really close.

**Comment by quicksketch** [comment]
This has been fixed in 1.x (1.5.0) through #1042 (converting this to a View), but that won't be in 1.4.x. I think I'll remove that test case for 1.4.x but leave it in 1.x, where this problem is fixed.

**Comment by quicksketch** [comment]
Okay I made a pair PRs. In 1.x we now have a better way of suppressing output than using exit(). So I made two PRs:

1.4.x: https://github.com/backdrop/backdrop/pull/1575
1.x: https://github.com/backdrop/backdrop/pull/1576

Both include the new test @serundeputy added (nice one). So we can check the problem is resolved. The added `exit()` breaks a test in 1.4.x, but no functionality is actually broken by it. So in 1.4.x I disabled that test. It's not affected in 1.x, where we have a better fix in place.
